linera-views-derive/src/lib.rs

```diff
@@ -21,61 +24,29 @@ fn get_seq_parameter(generics: syn::Generics) -> Vec<syn::Ident> {
     generic_vect
 }
 
-fn custom_attribute(attributes: &[Attribute], key: &str) -> Option<LitStr> {
-    attributes
-        .iter()
-        .filter(|attribute| attribute.path().is_ident("view"))
-        .filter_map(|attribute| match attribute.parse_args() {
-            Ok(MetaNameValue {
-                path,
-                value:
-                    syn::Expr::Lit(syn::ExprLit {
-                        lit: Lit::Str(value),
-                        ..
-                    }),
-                ..
-            }) => path.is_ident(key).then_some(value),
-            _ => panic!(
-                r#"Invalid `view` attribute syntax. \
-                Expected syntax: `#[view(key = "value")]`"#,
-            ),
-        })
-        .next()
-}
-
 fn context_and_constraints(
-    attributes: &[Attribute],
+    item: &syn::ItemStruct,
     template_vect: &[syn::Ident],
-) -> (Type, WhereClause) {
-    let context;
-    let constraints;
+) -> (Type, Vec<syn::WherePredicate>) {
+    let attrs: StructAttrs = deluxe::parse_attributes(item).unwrap();
```

Comment:
We do not like to put types like that.
Here we prefer
```rust
    let attrs = deluxe::parse_attributes::<_, StructAttrs>(item).unwrap();
```

Reply:
I'm not sure I like that (at least in this case) because it makes the use of the type variable much less clear. The fact we passed `StructAttrs` to `parse_attributes` doesn't tell us what it's used for, whereas by typing the variable we know that it's the type of the resulting variable.

Reply:
As you want.
As mentioned in the `CONTRIBUTE.md`, we put type annotations only when the compiler requires it, and if so, we prefer to do it in the right-hand side.